packages/frontend-api/src/Model/Advert/AdvertRepository.php

```diff
@@ -53,13 +55,23 @@ protected function getAdvertRepository()
     /**
      * @param int $domainId
      * @param string $positionName
+     * @param \Shopsys\FrameworkBundle\Model\Category\Category|null $category
      * @return \Doctrine\ORM\QueryBuilder
      */
-    protected function getVisibleAdvertsByPositionNameQueryBuilder(int $domainId, string $positionName)
+    protected function getVisibleAdvertsByPositionNameQueryBuilder(int $domainId, string $positionName, ?Category $category = null)
     {
-        return $this->getVisibleAdvertsQueryBuilder($domainId)
+        $queryBuilder = $this->getVisibleAdvertsQueryBuilder($domainId)
             ->andWhere('a.positionName = :positionName')
             ->setParameter('positionName', $positionName);
+
+        if ($category !== null) {
+            $queryBuilder
+                ->leftJoin('a.categories', 'c')
+                ->andWhere('c IN(:category) OR c IS NULL')
```

Comment:
Same comment as in framework/AdvertRepository in case that one gets accepted. :slightly_smiling_face: